unreachable code

latest release (11.1.1), MSC 19.42.34433 (VS2022 with latest updates) complains:

```
fmt\include\fmt\base.h(1538) : warning C4702: unreachable code
fmt\include\fmt\base.h(1542) : warning C4702: unreachable code
fmt\include\fmt\format.h(3425) : warning C4702: unreachable code

```

enable/disable FMT_NORETURN resp. [[noreturn]] has no effect.

```
if (end - fill_end <= 0) {
  report_error("invalid format specifier");
  return begin;
}

```
Do other compiler complain if `return begin;` is removed?

I've also got some unreachable code warnings with fmt 11.1.1 [CI log](https://github.com/GamesTrap/TRAP/actions/runs/12536938472/job/34960502512#step:11:1019):

```
C:\actions-runner\_work\TRAP\TRAP\Dependencies\fmt\include\fmt\format.h(3508) : error C2220: the following warning is treated as an error
C:\actions-runner\_work\TRAP\TRAP\Dependencies\fmt\include\fmt\format.h(2188) : warning C4702: unreachable code
C:\actions-runner\_work\TRAP\TRAP\Dependencies\fmt\include\fmt\base.h(893) : warning C4702: unreachable code
C:\actions-runner\_work\TRAP\TRAP\Dependencies\fmt\include\fmt\base.h(1324) : warning C4702: unreachable code
C:\actions-runner\_work\TRAP\TRAP\Dependencies\fmt\include\fmt\base.h(1408) : warning C4702: unreachable code
C:\actions-runner\_work\TRAP\TRAP\Dependencies\fmt\include\fmt\base.h(1538) : warning C4702: unreachable code
C:\actions-runner\_work\TRAP\TRAP\Dependencies\fmt\include\fmt\base.h(1542) : warning C4702: unreachable code
```

--- a/include/fmt/format-inl.h
+++ b/include/fmt/format-inl.h
@@ -134,7 +134,7 @@ FMT_FUNC auto write_loc(appender out, loc_value value,
 
 FMT_FUNC void report_error(const char* message) {
 #if FMT_USE_EXCEPTIONS
-  throw format_error(message);
+  FMT_THROW(format_error(message));
 #else
   fputs(message, stderr);
   abort();

